src/main/java/frc/robot/subsystems/swerve/SwerveCommands.java

```diff
@@ -28,15 +30,46 @@ public class SwerveCommands {
 
     private static final Swerve SWERVE = RobotContainer.SWERVE;
 
+    private static final SysIdCalibrator STEER_CALIBRATOR = new SysIdCalibrator(
+            true,
+            SWERVE,
+            SWERVE::runModuleSteerByVoltage,
+            2
+    );
+
+    private static final SysIdCalibrator DRIVE_CALIBRATOR = new SysIdCalibrator(
+            true,
+            SWERVE,
+            SWERVE::runModulesDriveByVoltage,
+            2
+    );
+
+    public static Command getSteerCalibration(boolean isQuasistatic, SysIdRoutine.Direction direction) {
+        //todo - add coupling handel
+        Command command = STEER_CALIBRATOR.getSysIdCommand(isQuasistatic, direction);
+        command.setName("Steer Calibration");
+        return command;
+    }
+
+    // Must start when all wheels looking forward
+    public static Command getDriveCalibration(boolean isQuasistatic, SysIdRoutine.Direction direction) {
+        Command command = DRIVE_CALIBRATOR.getSysIdCommand(isQuasistatic, direction);
+        command.setName("Drive Calibration");
+        return command;
+    }
+
     public static Command getWheelRadiusCalibrationCommand() {
-        Command command = new WheelRadiusCharacterization(
-                SWERVE,
-                SwerveConstants.DRIVE_RADIUS_METERS,
-                Rotation2d.fromRotations(0.5),
-                SWERVE::getModulesDriveDistances,
-                SWERVE::getAbsoluteHeading,
-                SWERVE::runWheelRadiusCharacterization,
-                SWERVE::stop
+        Command command = new SequentialCommandGroup(
+                getReadySpinSwerveCommand(),
+                new WheelRadiusCharacterization(
+                        SWERVE,
+                        SwerveConstants.DRIVE_RADIUS_METERS,
+                        Rotation2d.fromRotations(0.5),
```

Comment:
half circle constant

Reply:
The number here 0.5 used for velocity and can be change and not for calculation so can’t use half circle but I can do another const